lotto.py

```
@@ -0,0 +1,692 @@
+    if len(token) != 64:
+        raise HTTPException(status_code=400, detail=f"{field_name} must be 32-byte hex")
+    if any(ch not in string.hexdigits for ch in token):
+        raise HTTPException(status_code=400, detail=f"{field_name} must be hex")
+    return token
+
+
+def _parse_int(value: object, field_name: str) -> int:
+    try:
+        return int(str(value))
+    except (TypeError, ValueError):
+        raise HTTPException(status_code=400, detail=f"{field_name} must be an integer")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)